### PR TITLE
feat: add datetime block [APE-734]

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

Add some linked data to `ReceiptAPI` and extra processing on `BlockAPI.timestamp` to get a datetime obj

fixes: #1257

### How I did it

Uses `ChainManager.blocks` to fetch the blocks in order to take advantage of the query manager

### How to verify it

Call the methods directly and via `AccountHistoryContainer.query`/`BlockContainer.query` to verify if any issues are introduced

### Checklist

- [x] All changes are completed
- [x] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

File: src/ape/api/providers.py [hunk 1]
```diff
@@ -84,6 +85,10 @@ class BlockAPI(BaseInterfaceModel):
     size: int
     timestamp: int
 
+    @property
+    def datetime(self) -> datetime:
+        return datetime.fromtimestamp(self.timestamp)
```
[the remark this answers is not in the document]
Comment (unparalleled-js) [reply src/ape/api/providers.py:90]: Should this be `utcfromtimestamp` ?

Comment (unparalleled-js) [inline src/ape/api/providers.py:90]: could have a test showing this `block.datetime.timestamp == block.timestamp`

[the remark this answers is not in the document]
Comment (fubuloubu) [reply src/ape/api/providers.py]: suggestion is to use `tz=` kwarg instead

**Comment by fubuloubu** [comment]
NOTE: Validated against etherscan for `BlockAPI.datetime` and `TransactionAPI.datetime` matches what it displays (in UTC)

**Comment by fubuloubu** [comment]
NOTE: can convert to Pandas `Timestamp` object via `pandas.to_datetime(df.datetime)`

File: tests/functional/test_receipt.py [hunk 2]
```diff
@@ -10,6 +10,12 @@ def invoke_receipt(vyper_contract_instance, owner):
     return vyper_contract_instance.setNumber(1, sender=owner)
 
 
+def test_receipt_properties(chain, invoke_receipt):
+    assert chain.blocks.head.number == invoke_receipt.block_number
```
Comment (unparalleled-js) [inline tests/functional/test_receipt.py:14]: can you flip these assertions so the "actual" (receipt) comes first?

Comment (fubuloubu) [inline tests/functional/test_receipt.py:14]:
```suggestion
    assert invoke_receipt.block_number == chain.blocks.head.number
```

File: tests/functional/test_receipt.py [hunk 3]
```diff
@@ -10,6 +10,12 @@ def invoke_receipt(vyper_contract_instance, owner):
     return vyper_contract_instance.setNumber(1, sender=owner)
 
 
+def test_receipt_properties(chain, invoke_receipt):
+    assert invoke_receipt.block_number == chain.blocks.head.number
+    assert chain.blocks.head.timestamp == invoke_receipt.timestamp
+    assert chain.blocks.head.datetime == invoke_receipt.datetime
```
Comment (fubuloubu) [inline tests/functional/test_receipt.py:16]:
```suggestion
    assert invoke_receipt.timestamp == chain.blocks.head.timestamp
    assert invoke_receipt.datetime == chain.blocks.head.datetime
```

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**